src/solvers/scip.rs

```diff
@@ -90,6 +195,64 @@ impl SCIPProblem {
     pub fn as_inner_mut(&mut self) -> &mut Model<ProblemCreated> {
         &mut self.model
     }
+
+    /// Sets whether or not SCIP should display verbose logging information to the console
+    pub fn try_set_verbose(mut self, verbose: bool) -> Result<Self, Retcode> {
+        self.model = std::mem::take(self.as_inner_mut())
+            .set_int_param("display/verblevel", if verbose { 4 } else { 0 })?;
+        Ok(self)
+    }
+
+    /// Tries to set whether or not SCIP should display verbose logging
+    /// information to the console and panics if the operation fails
+    pub fn set_verbose(self, verbose: bool) -> Self {
+        self.try_set_verbose(verbose)
+            .unwrap_or_else(|e| panic!("cound not set verbosity to {}: {:?}", verbose, e))
+    }
+
+    /// Sets the heuristics parameter of the SCIP instance
+    pub fn set_heuristics(mut self, heuristics: ScipHeuristics) -> Self {
+        self.model = std::mem::take(self.as_inner_mut()).set_heuristics(match heuristics {
+            ScipHeuristics::Default => russcip::ParamSetting::Default,
+            ScipHeuristics::Aggressive => russcip::ParamSetting::Aggressive,
+            ScipHeuristics::Fast => russcip::ParamSetting::Fast,
+            ScipHeuristics::Off => russcip::ParamSetting::Off,
+        });
+        self
+    }
+
+    /// Sets the time limit in seconds
+    pub fn set_time_limit(mut self, time_limit: usize) -> Self {
+        self.model = std::mem::take(self.as_inner_mut()).set_time_limit(time_limit);
+        self
+    }
+
+    /// Sets the memory limit in MB
+    pub fn set_memory_limit(mut self, memory_limit: usize) -> Self {
+        self.model = std::mem::take(self.as_inner_mut()).set_memory_limit(memory_limit);
+        self
+    }
+
+    /// Sets a SCIP parameter
+    pub fn try_set_option<T: ScipOptionSetValue>(
+        mut self,
+        option: &str,
+        value: T,
+    ) -> Result<Self, Retcode> {
+        self.model = value.set_for(std::mem::take(self.as_inner_mut()), option)?;
```

Comment:
are you sure this is what you want to do? You are silently dropping all attributes in self except model

Reply:
Erm … then I am probably still skill-issuing myself here regarding `std::mem::take`. How else would this be solved?

Reply:
If it's easier to fix this than to explain it to me, feel free to go ahead and push to my branch